doc/cpp/cable_cell_format.rst

```diff
@@ -0,0 +1,88 @@
+.. _cppcablecellformat:
+
+Description Format
+==================
+
+Arbor provides readers and writers for describing :ref:`label dictionaries <labels>`,
+:ref:`decoration objects <cablecell-decoration>`, :ref:`morphologies <morph>` and
+:ref:`cable cells <cablecell>`, referred to here as *arbor-components*.
+
+A detailed description of the s-expression format used to describe each of these components
+can be found :ref:`here <formatcablecell>`.
+
+Reading and writing of the arbor-component description format is delegated to the ``arborio``
+library and the responsible classes and functions are present in the ``arborio`` namespace.
+
+The arbor-components and meta-data
+----------------------------------
+
+.. cpp:type:: cable_cell_variant = std::variant<arb::morphology, arb::label_dict, arb::decor, arb::cable_cell>
+
+.. cpp:type:: template <typename T> parse_hopefully = arb::util::expected<T, cableio_parse_error>
+
+   ``arborio::cableio_parse_error`` is derived from ``arb::arbor_exception`` which in turn is derived
+   from ``std::runtime_error``. It contains information about the ``line`` and ``position`` of an encountered
+   error in a document.
+
+   ``arb::util::expected`` contains either an object of type ``T`` or an error object.
+
+.. cpp:class:: meta_data
+
+   .. cpp:member:: int version
```

Comment:
Shouldn't this be capable of holding '0.1-dev'?

Reply:
Yes, thanks :)

Reply:
👍